### DISCUSSION TITLE
[BUG] Error parsing called workflow in reusable workflow

### DISCUSSION THREAD

**Discussion by daniel-butler-irl** [community/community#55649] [Actions]
### Select Topic Area

Bug

### Body

When utilizing a reusable workflow in GitHub Actions, an error occurs during parsing of the called workflow. The exact error message received is as follows:
```bash
[Invalid workflow file: .github/workflows/main.yml#L10](https://github.com/daniel-test-bed/pipeline-triggers/actions/runs/4993357570/workflow)
error parsing called workflow
".github/workflows/main.yml"
-> "daniel-test-bed/workflows/.github/workflows/wf-wrapper.yml@v1.2.21" (source tag with sha:c99021debcd09aec9ccd8d4e285c3572738d947d)
--> "./.github/workflows/set-commit-status.yml"
: workflow was not found. See https://docs.github.com/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows for more information.
```
the workflow its self is basic
```YAML
name: CI
on:
  issue_comment:
    types:
      - created
jobs:
  mustpass:
    name: Job Must Pass
    uses: daniel-test-bed/workflows/.github/workflows/wf-wrapper.yml@v1.2.21
    secrets: inherit
    with:
      status-name: "CI/Job Must Pass"
      name: ${{ github.event.comment.user.login }}
```

The issue arises when attempting to access the reusable workflow with the tag v1.2.21, which is signed and associated with the SHA c99021debcd09aec9ccd8d4e285c3572738d947d. However, instead of following the signed tag to the target SHA 9cb81cd5c7fb7b2c93a03c6a4ce3e37e050662ac, the workflow just tries to use the tag which does not contain the code. I think this is what is happening anyway...  

```bash
❯ git show-ref --dereference v1.2.21
c99021debcd09aec9ccd8d4e285c3572738d947d refs/tags/v1.2.21
9cb81cd5c7fb7b2c93a03c6a4ce3e37e050662ac refs/tags/v1.2.21^{}

```

If I sign the tag on creation I get no symbolic link and everything works. However, a completely separate issue has me signing the tag after creation; this is where I see the bug in the workflow.

**Comment by luismendezescobar** [comment]
same issue here, has anyone found a solution?

**Comment by sakojun** [comment]
I was also having the same error, but it has been resolved.
It seems that this can happen when using annotated tags.
The solution is not to use annotated tags.

**Reply by stvenx** [reply]
I encountered the same error. When I create a tag using the Git CLI, the Actions get the tag's sign but not the commit SHA. I used the GitHub website to release the tag and fix this problem.

**Reply by troydwhcc** [reply]
I was able to resolve this by making the caller, and called repository internal.

1. Concerning private and internal repositories:
- A private can reference a private repo. An internal can reference an internal repo. An internal can reference a private. A private _cannot_ reference an internal repo.

2. The called repository must allow access from within the organization or enterprise.
3. The calling repository "Action Permissions" must allow reusable workflows outside of the repository.
